### DISCUSSION TITLE
using onBlur function

### DISCUSSION THREAD

**Discussion by Mood-al** [react-hook-form/react-hook-form#6410] [General]
Hey guys i'm using react-hook-form 7 
but I'm having troubles with onBlur function if i called it like so inside my input 
` 
<input
                 
   // dir={isRTL(watch("name") || donor?.name) ? "rtl" : "ltr"}
                  
  onFocus={onNameFocus}
                    onBlur={myBlurFunction}
                    dir={dir}
                    className="form-control"
                    type="text"
                    placeholder={t(`name`)}
                    {...register("name")}
                    defaultValue={donor?.name}
                    required
                  />`
![image](https://user-images.githubusercontent.com/63562308/131517797-089d2e19-b007-4045-9d4c-2ec592fa227b.png)

it doesnt work or return anything i want it to change the direction of my input after bluring 
is there any possible way to use it ?

**Comment by Mood-al** [comment]
I figured it out after 3 hours of working 😒 
i should put the register method above all the code right under input tag i dont know why this bahavior happened it's really odd 
![image](https://user-images.githubusercontent.com/63562308/131533814-0a57fdd8-758d-46d8-92a5-2fcef9646fe2.png)

**Comment by bluebill1049** [comment]
have a read on this section: 

Custom onChange, onBlur

When you want to combine with your onChange, onBlur, you can achieve by the following:

```tsx
// onChange got overwrite by register method
<input onChange={handleChange} {...register('test')} />

// register's onChange got overwrite by register method
<input {...register('test')} onChange={handleChange}/>

const firstName = register('firstName', { required: true })
<input 
  onChange={(e) => {
    firstName.onChange(e); // method from hook form register
    handleChange(e); // your method
  }}
  onBlur={firstName.onBlur}
  ref={firstName.ref} 
/>
```

https://react-hook-form.com/api/useform/register

<img width="966" alt="Screen Shot 2021-09-01 at 9 05 53 am" src="https://user-images.githubusercontent.com/10513364/131586926-291befab-b5a1-4ca5-b6d7-532101b71847.png">
